File: delete_wo_pointer.py
# Time Complexity : Add - O(1)
# Space Complexity :O(1)
# Did this code successfully run on Leetcode : Yes
# Any problem you faced while coding this : No
'''
1. I have used Swapping to copying data of nextodelete node, and adress its poiinting to
'''

def deleteNode(curr_node):
    #code here
    temp = curr_node.next
    curr_node.data = temp.data
    curr_node.next = temp.next
    temp = None

File: DeleteWoPointer.java
// Time Complexity : O(1)
// Space Complexity : O(1)
// Did this code successfully run on Leetcode : No
// Any problem you faced while coding this :


// Your code here along with comments explaining your approach
// Copying the next element to the current and point to the next.next
// if there is the last element, auto assign as null
public class Solution {
    public void deleteNode(ListNode pos) {
        if(pos.next != null){
            pos.val = pos.next.val;
            pos.next = pos.next.next;
        }else{
            pos = null;
        }
    }
}